src/app/products/_components/filter/Filter.tsx

```diff
@@ -0,0 +1,71 @@
+'use client';
+
+import React, { useState } from 'react';
+
+import { PriceRange, FilterProps } from '../../../../types/product';
+import { useFilterOptions } from '@/hooks/useFilterOptions';
+
+import { SelectedOptionTag } from '@/app/products/_components/filter/SelectedOptionTag';
+import { PriceFilter } from '@/app/products/_components/filter/PriceFilter';
+import { ColorFilter } from '@/app/products/_components/filter/ColorFilter';
+
+const Filter: React.FC<FilterProps> = ({ products }) => {
+  const [priceRange, setPriceRange] = useState<PriceRange>(() => ({
+    min: Math.min(...products.map((p) => p.price)),
+    max: Math.max(...products.map((p) => p.price)),
+  }));
+  const [selectedPriceRange, setSelectedPriceRange] = useState<PriceRange | undefined>(undefined);
+  const availableOptions = useFilterOptions(products);
+  const [sliderValue, setSliderValue] = useState([priceRange.min, priceRange.max]);
+  const [selectedColors, setSelectedColors] = useState<string[]>([]);
+
+  const handlePriceSearch = () => setSelectedPriceRange(priceRange);
+  const handleSliderChange = (value: number[]) => {
+    setSliderValue(value);
+    setPriceRange({ min: value[0], max: value[1] });
+  };
```

Comment:
```suggestion
  const handleSliderChange = ([min, max]: [number, number]) => {
    setSliderValue([min, max]);
    setPriceRange({ min, max });
  };
```

혹시 이렇게 짜는건 어떠신가요? value에 [number, number] 형식의 값이 들어오는 것 같아 제안드립니다!